lib/public_suffix.ex

```diff
@@ -59,13 +59,63 @@ defmodule PublicSuffix do
   @spec registrable_domain(String.t, options) :: nil | String.t
   def registrable_domain(domain, options \\ []) when is_binary(domain) do
     # "The registered or registrable domain is the public suffix plus one additional label."
-    parse_domain(domain, options, 1)
+    parse_domain(domain, options, 1, :registrable_or_suffix)
+  end
+
+  @doc """
+  Parses the provided domain and returns the prevailing rule based on the
+  publicsuffix.org rules. If no rules match, the prevailing rule is "*".
+
+  ## Examples
+
+      iex> prevailing_rule("foo.bar.com")
+      "com"
+      iex> prevailing_rule("co.uk")
+      "co.uk"
+      iex> prevailing_rule("foobar.example")
+      "*"
+
+  You can use the `ignore_private` keyword to exclude private (non-ICANN) domains.
+
+      iex> prevailing_rule("foo.github.io", ignore_private: false)
+      "github.io"
+      iex> prevailing_rule("foo.github.io", ignore_private: true)
+      "io"
+      iex> prevailing_rule("foo.github.io")
+      "github.io"
+  """
+  @spec prevailing_rule(String.t) :: nil | String.t
+  @spec prevailing_rule(String.t, options) :: nil | String.t
+  def prevailing_rule(domain, options \\ []) when is_binary(domain) do
+    parse_domain(domain, options, [], :prevailing_rule)
+  end
+
+  @doc """
+  Checks whether the provided domain matches an existing rule in the
+  publicsuffix.org rules.
+
+  ## Examples
+      iex> matches_explicit_rule?("foo.bar.com")
+      true
+      iex> matches_explicit_rule?("com")
+      true
+      iex> matches_explicit_rule?("foobar.example")
+      false
+  """
+  @spec matches_explicit_rule?(String.t | nil) :: boolean
+  def matches_explicit_rule?(nil), do: false
+  def matches_explicit_rule?(domain) when is_binary(domain) do
```

Comment:
I was thinking about that, but wondered whether it would matter - is it possible that there could be a private domain that would _not_ match an explicit non-private rule if `ignore_private: true` was passed?